benchmark/CacheMap.hs

```diff
@@ -40,9 +53,35 @@ tenLookups tmap = (lp, lp, lp, lp, lp, lp, lp, lp)
     lp :: forall (a::Nat). Typeable a => Proxy a
     lp = fromJust $ lookup tmap
 
+inserts :: forall a . (KnownNat a)
+        => TypeRepMap (Proxy :: Nat -> *)
+        -> Int
+        -> Proxy (a :: Nat)
+        -> TypeRepMap (Proxy :: Nat -> *)
+inserts !c 0 _ = c
+inserts !c n x = inserts
+   (insert x c)
+   (n-1)
+   (Proxy :: Proxy (a+1))
+
+updates :: forall a . (KnownNat a)
+        => TypeRepMap (Proxy :: Nat -> *)
+        -> Int
+        -> Proxy (a :: Nat)
+        -> TypeRepMap (Proxy :: Nat -> *)
+updates !c 0 _ = c
+updates !c n x = inserts
+   (insert x c)
+   (n-1)
+   (Proxy :: Proxy (a+1))
+
+mkSmallMap :: IO (TypeRepMap (Proxy :: Nat -> *))
+mkSmallMap = pure $ fromList $ buildBigMap 10 (Proxy :: Proxy 0) []
```

Comment:
Why it needs to be wrapped inside `IO`? :thinking:

Reply:
because of the `env` type